src/test/java/uk/ac/ebi/embl/template/TemplateEntryProcessorTest.java

```diff
@@ -0,0 +1,180 @@
+import uk.ac.ebi.embl.api.validation.helper.FlatFileComparatorException;
+import uk.ac.ebi.embl.api.validation.submission.SubmissionOptions;
+import uk.ac.ebi.embl.flatfile.reader.genomeassembly.ChromosomeListFileReaderTest;
+import uk.ac.ebi.ena.webin.cli.validator.reference.Sample;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.sql.Connection;
+import java.util.*;
+
+import static org.junit.Assert.*;
+
+public class TemplateEntryProcessorTest {
+
+    private final static String AUTH_JSON="{\"authRealms\":[\"ENA\"],\"password\":\"sausages\",\"username\":\"Webin-256\"}";
+    private final static String TEST_AUTH_URL="https://wwwdev.ebi.ac.uk/ena/submit/webin/auth/token";
+    private final static File templateFile = Paths.get(System.getProperty("user.dir") + "/src/test/resources/templates/ERT000002.xml").toFile();
+    private final static String MOL_TYPE = "/mol_type";
+    private String token="";
+    private TemplateInfo templateInfo ;
+    private TemplateEntryProcessor templateEntryProcessor;
+    private TemplateVariables templateVariables;
+    private String molType;
+    
+    
+    @Before
+    public void setUp() throws Exception {
+        templateInfo = new TemplateLoader().loadTemplateFromFile(templateFile);
+        templateEntryProcessor = getTemplateEntryProcessor();
+        molType = getMolTypeFromTemplateForTest(templateInfo);
+    }
+    
+
+    @Test
+    public void  testEntryProcess() throws Exception {
+
+        // Test with biosample Id
+        executeEntryProcessTest("SAMEA9403245");
+        
+        // Test with sample Id
+        executeEntryProcessTest("ERS7118926");
+        
+        // Test with sample alias "test_custom"
+        executeEntryProcessTest("test_custom");
+
+        // Test with valid organism
+        executeEntryProcessTestWithScientificName("Homo sapiens");
+
+        // Test with tacId
+        executeEntryProcessTestWithTaxId("9606");
+
+        // Test with invalid organizm
+        executeEntryProcessInvalidOrganism("JUNK");
+
+    }
+    
+    public void executeEntryProcessTest(String organismName) throws Exception{
+        
+        TemplateVariables templateVariables = getTemplateVariables(organismName);
+        Sample sample=getSampleForTest(templateEntryProcessor,templateVariables);
+        
+        TemplateProcessorResultSet templateProcessorResultSet = templateEntryProcessor.processEntry(templateInfo, molType, templateVariables,getOptions());
+
+        SourceFeature sourceFeature=templateProcessorResultSet.getEntry().getPrimarySourceFeature();
+        assertTrue(templateProcessorResultSet.getValidationResult().isValid());
+        assertEquals(sourceFeature.getTaxId(),Long.valueOf(sample.getTaxId()));
+        assertEquals(sourceFeature.getScientificName(),sample.getOrganism());
+    }
+
+    public void executeEntryProcessTestWithScientificName(String taxId) throws Exception{
```

Comment:
- Suggestion is to pass in the templateInfo and molType so that these test methods are template independent.
- Change taxId to scientificName.